src/cli/commands/install.js

```diff
@@ -947,6 +947,17 @@ export async function run(config: Config, reporter: Reporter, flags: Object, arg
   if (flags.lockfile === false) {
     lockfile = new Lockfile();
   } else {
+    const haveLockfile = await fs.exists(path.join(config.lockfileFolder, constants.LOCKFILE_FILENAME));
+    if (haveLockfile) {
+      //no lockfile found, test if node_module exists
+      const haveNodeModules = await fs.exists(path.join(config.lockfileFolder, constants.NODE_MODULES_FOLDER));
```

Comment:
I don't see the benefit of checking the existence of this folder only after checking for the lockfile. Since these are both async functions, doing these side by side would probably be more performant.

Reply:
agree, thank you.